app/controllers/hosts_controller.rb

```diff
@@ -416,6 +417,34 @@ def multiple_destroy
   def multiple_build
   end
 
+  def rebuild_config
+  end
+
+  def submit_rebuild_config
+    all_fails = {}
+    @hosts.each do |host|
+      result = host.recreate_config
+      result.each_pair do |k, v|
+        all_fails[k] = all_fails[k] || []
+        all_fails[k] << host.name unless v
+      end
+    end
+
+    message = ''
+    all_fails.each_pair do |key, values|
+      unless values.empty?
+        message << ((_("%{config_type} rebuild failed for hosts: %{host_names}.") % {:config_type => key.to_s.upcase, :host_names => values.to_sentence})) + " "
```

Comment:
I don't think you need .upcase here, if the key is actually the "pretty name".  If it is the pretty name then I think you can do `_(key)` to get the localised version.